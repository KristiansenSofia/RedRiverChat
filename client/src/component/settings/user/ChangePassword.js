import React, { Component } from 'react';

// Import NPM-modules
import Typography from 'material-ui/Typography';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import axios from 'axios';

// Import styles. settingsUserStyles for all imported components with a style attribute and CSS-file for classNames and id.
import {settingsUserStyles} from "../../../styles/SettingsStyles";
import '../../../styles/Styles.css'
import {registerStyles} from "../../../styles/AuthStyles";

/**
 *  Change Password-component.
 *
 *  @author Jimmy
 */

class ChangePassword extends Component {

    constructor(props){
        super(props);

        this.state = {
            password: '',
            passwordConfirm: '',
        };

    }

    /**
     *  Handle form-input. Inputs are added to this.state.
     *
     *  @author Jimmy
     */

    handleChange = name => event => {
        this.setState({
            [name]: event.target.value,
        });
    };


    /**
     *  Handle submit-button for change password
     *
     *  @author Jimmy
     */

    handleSubmit() {

        if (this.state.password === '' || this.state.passwordConfirm === '') {
            return this.props.openSnackBar('Formuläret ej korrekt ifyllt!');
        }

        if (this.state.password !== this.state.passwordConfirm) {
            return this.props.openSnackBar('Lösenorden matchar inte!');
        }

        // ToDo.. Add this when function to change users password is implemented on server.
        /*this.sendRequest()
            .then((response) => {

                console.log(response);
                this.setState({navigate: true,});
                return this.props.openSnackBar('Registreringen lyckades. Vänligen logga in!');

            }).catch((err) => {
            console.log(err);
            return this.props.openSnackBar('Något gick fel. Försök igen!');
        });*/

    }

    render() {
        return (
            <div className="UserDetails">
                <form style={settingsUserStyles.container} noValidate autoComplete="off">
                    <TextField
                        id="password"
                        label="Nytt lösenord"
                        style={settingsUserStyles.textField}
                        type="password"
                        autoComplete="current-password"
                        onChange={this.handleChange('password')}
                        margin="normal"
                    />
                    <TextField
                        id="passwordRepeat"
                        label="Bekräfta lösenord"
                        style={settingsUserStyles.textField}
                        type="password"
                        autoComplete="current-password"
                        onChange={this.handleChange('passwordConfirm')}
                        margin="normal"
                    />
                        <Button variant="raised" style={settingsUserStyles.button} onClick={this.handleSubmit}>
                            Ok
                        </Button>
                </form>
            </div>
        );
    }
}

export default ChangePassword;