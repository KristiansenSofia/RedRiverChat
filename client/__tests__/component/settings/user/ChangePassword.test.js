import React from 'react';
import Enzyme from 'enzyme';
import {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ChangePassword from '../../../../src/component/settings/user/ChangePassword';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

Enzyme.configure({ adapter: new Adapter() });

describe('ChangePassword', () => {
  const props = {
    openSnackBar: jest.fn()
  };

  const component = shallow(
    <ChangePassword {...props}/>
  );

  it('renders without crashing', () => {
    expect(component).toMatchSnapshot();
  });


  describe('handleChange', () => {
    let event = {
      target: {
        value: 'Password123'
      }
    }

    it('should set password to Password123', () => {
      component.instance().handleChange('password')(event);
      const passwordState = component.state('password');

      expect(passwordState).toEqual('Password123');
    });

    it('should set passwordConfirm to Password123', () => {
      component.instance().handleChange('passwordConfirm')(event);
      const passwordConfirmState = component.state('passwordConfirm');

      expect(passwordConfirmState).toEqual('Password123');
    });
  });


  describe('handleSubmit', () => {
    it('should return "Formuläret ej korrekt ifyllt!" when password is missing', (done) => {
      const props1 = {
        openSnackBar: jest.fn()
      };

      const component1 = shallow(
        <ChangePassword {...props1}/>
      );

      component1.setState({
        password: '',
        passwordConfirm: 'Password123'
      });

      component1.instance().handleSubmit();

      expect(props1.openSnackBar).toHaveBeenCalledTimes(1);
      done();
    });

    it('should return "Formuläret ej korrekt ifyllt!" when passwordConfirm is missing', (done) => {
      const props2 = {
        openSnackBar: jest.fn()
      };

      const component2 = shallow(
        <ChangePassword {...props2}/>
      );

      component2.setState({
        password: 'Password123',
        passwordConfirm: ''
      });

      component2.instance().handleSubmit();

      expect(props2.openSnackBar).toHaveBeenCalledTimes(1);
      done();
    });

    it('should return "Lösenorden matchar inte!" when password and passwordConfirm are different', (done) => {
      const props3 = {
        openSnackBar: jest.fn()
      };

      const component3 = shallow(
        <ChangePassword {...props3}/>
      );

      component3.setState({
        password: 'Password123',
        passwordConfirm: 'password1'
      });

      component3.instance().handleSubmit();

      expect(props3.openSnackBar).toHaveBeenCalledTimes(1);
      done();
    });

    //TODO: Rewrite this test when sendRequest is implemented.
    it('should call sendRequest', (done) => {
      const props4 = {
        openSnackBar: jest.fn()
      };

      const component4 = shallow(
        <ChangePassword {...props4}/>
      );

      component4.setState({
        password: 'Password123',
        passwordConfirm: 'Password123'
      });

      component4.instance().handleSubmit();

      expect(props4.openSnackBar).toHaveBeenCalledTimes(0);
      done();
    });
  });
});
