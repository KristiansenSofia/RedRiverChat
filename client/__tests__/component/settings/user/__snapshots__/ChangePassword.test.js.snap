// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChangePassword
    openSnackBar={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        autoComplete="off"
        noValidate={true}
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "space-evenly",
          }
        }
      >
        <TextField
          autoComplete="current-password"
          id="password"
          label="Nytt lösenord"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          style={
            Object {
              "margin": 10,
            }
          }
          type="password"
        />
        <TextField
          autoComplete="current-password"
          id="passwordRepeat"
          label="Bekräfta lösenord"
          margin="normal"
          onChange={[Function]}
          required={false}
          select={false}
          style={
            Object {
              "margin": 10,
            }
          }
          type="password"
        />
        <WithStyles(Button)
          onClick={[Function]}
          style={
            Object {
              "margin": 20,
            }
          }
          variant="raised"
        >
          Ok
        </WithStyles(Button)>
      </form>,
      "className": "UserDetails",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "autoComplete": "off",
        "children": Array [
          <TextField
            autoComplete="current-password"
            id="password"
            label="Nytt lösenord"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            style={
              Object {
                "margin": 10,
              }
            }
            type="password"
          />,
          <TextField
            autoComplete="current-password"
            id="passwordRepeat"
            label="Bekräfta lösenord"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            style={
              Object {
                "margin": 10,
              }
            }
            type="password"
          />,
          <WithStyles(Button)
            onClick={[Function]}
            style={
              Object {
                "margin": 20,
              }
            }
            variant="raised"
          >
            Ok
          </WithStyles(Button)>,
        ],
        "noValidate": true,
        "style": Object {
          "display": "flex",
          "flexWrap": "wrap",
          "justifyContent": "space-evenly",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "autoComplete": "current-password",
            "id": "password",
            "label": "Nytt lösenord",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "style": Object {
              "margin": 10,
            },
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "autoComplete": "current-password",
            "id": "passwordRepeat",
            "label": "Bekräfta lösenord",
            "margin": "normal",
            "onChange": [Function],
            "required": false,
            "select": false,
            "style": Object {
              "margin": 10,
            },
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Ok",
            "onClick": [Function],
            "style": Object {
              "margin": 20,
            },
            "variant": "raised",
          },
          "ref": null,
          "rendered": "Ok",
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          autoComplete="off"
          noValidate={true}
          style={
            Object {
              "display": "flex",
              "flexWrap": "wrap",
              "justifyContent": "space-evenly",
            }
          }
        >
          <TextField
            autoComplete="current-password"
            id="password"
            label="Nytt lösenord"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            style={
              Object {
                "margin": 10,
              }
            }
            type="password"
          />
          <TextField
            autoComplete="current-password"
            id="passwordRepeat"
            label="Bekräfta lösenord"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            style={
              Object {
                "margin": 10,
              }
            }
            type="password"
          />
          <WithStyles(Button)
            onClick={[Function]}
            style={
              Object {
                "margin": 20,
              }
            }
            variant="raised"
          >
            Ok
          </WithStyles(Button)>
        </form>,
        "className": "UserDetails",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "autoComplete": "off",
          "children": Array [
            <TextField
              autoComplete="current-password"
              id="password"
              label="Nytt lösenord"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              style={
                Object {
                  "margin": 10,
                }
              }
              type="password"
            />,
            <TextField
              autoComplete="current-password"
              id="passwordRepeat"
              label="Bekräfta lösenord"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              style={
                Object {
                  "margin": 10,
                }
              }
              type="password"
            />,
            <WithStyles(Button)
              onClick={[Function]}
              style={
                Object {
                  "margin": 20,
                }
              }
              variant="raised"
            >
              Ok
            </WithStyles(Button)>,
          ],
          "noValidate": true,
          "style": Object {
            "display": "flex",
            "flexWrap": "wrap",
            "justifyContent": "space-evenly",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "autoComplete": "current-password",
              "id": "password",
              "label": "Nytt lösenord",
              "margin": "normal",
              "onChange": [Function],
              "required": false,
              "select": false,
              "style": Object {
                "margin": 10,
              },
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "autoComplete": "current-password",
              "id": "passwordRepeat",
              "label": "Bekräfta lösenord",
              "margin": "normal",
              "onChange": [Function],
              "required": false,
              "select": false,
              "style": Object {
                "margin": 10,
              },
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Ok",
              "onClick": [Function],
              "style": Object {
                "margin": 20,
              },
              "variant": "raised",
            },
            "ref": null,
            "rendered": "Ok",
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
